release-note-script/src/main/java/MergeReleaseNotes.java

```diff
@@ -0,0 +1,240 @@
+
+      File file = new File(path);
+      load(edition, repository, file);
+    }
+    output();
+  }
+
+  void load(Edition edition, Repository repository, File file) throws Exception {
+    Category category = null;
+    String line;
+    Matcher matcher;
+
+    try (BufferedReader reader = new BufferedReader(new FileReader(file, StandardCharsets.UTF_8))) {
+      while ((line = reader.readLine()) != null) {
+        if (DEBUG != null) System.err.printf("ReadLine: %s%n", line);
+        matcher = Pattern.compile("^## *(\\p{Print}+) *$").matcher(line);
+        if (matcher.matches()) {
+          if (DEBUG != null) System.err.printf("matched: %s%n", matcher.group(1));
+          if (!SECTION_SUMMARY.equalsIgnoreCase(matcher.group(1)))
+            category = Category.getByDisplayName(matcher.group(1));
+          continue;
+        }
+
+        matcher = Pattern.compile("^ *- *(\\p{Print}+)$").matcher(line);
+        if (matcher.matches()) {
+          if (category == null)
+            throw new IllegalStateException(
+                "Missing category. Release note text: " + matcher.group(1));
+          addReleaseNote(new ReleaseNote(edition, category, repository, matcher.group(1)));
+        }
+      }
+    }
+  }
+
+  private void addReleaseNote(ReleaseNote releaseNote) {
+    Map<Category, Map<Repository, List<ReleaseNote>>> categoryMap =
+        editionMap.computeIfAbsent(releaseNote.edition, k -> new EnumMap<>(Category.class));
+
+    Map<Repository, List<ReleaseNote>> repositoryMap =
+        categoryMap.computeIfAbsent(releaseNote.category, k -> new EnumMap<>(Repository.class));
+
+    List<ReleaseNote> releaseNotesList =
+        repositoryMap.computeIfAbsent(releaseNote.repository, k -> new ArrayList<>());
+
+    if (releaseNote.edition.equals(Edition.ENTERPRISE)) {
+      Matcher matcher =
+          Pattern.compile("(.*) +(\\((#[0-9]+ *)+\\))$").matcher(releaseNote.releaseNoteText);
+      if (matcher.matches()) {
+        if (DEBUG != null)
+          System.err.printf(
+              "Matched::%s::%s grp1:%s grp2:%s%n",
+              releaseNote.category, releaseNote.repository, matcher.group(1), matcher.group(2));
+        releaseNote.releaseNoteText = matcher.group(1);
+      }
+    }
+
+    releaseNotesList.add(releaseNote);
+    repositoryMap.put(releaseNote.repository, releaseNotesList);
+    categoryMap.put(releaseNote.category, repositoryMap);
+    editionMap.put(releaseNote.edition, categoryMap);
+  }
+
+  void output() {
+    System.out.print("## Summary\n\n");
+    Arrays.stream(Edition.values())
+        .forEach(
+            edition -> {
+              outputSections(edition);
+              System.out.println();
+            });
+  }
+
+  private void outputSections(Edition edition) {
+    Map<Category, Map<Repository, List<ReleaseNote>>> categoryMap = editionMap.get(edition);
+    if (categoryMap == null || categoryMap.isEmpty()) return;
+
+    System.out.printf("## %s edition%n", edition.getEdition());
```

Comment:
Yes. when I use formatted output method like `printf`, the static analyzer (SpotBugs, ErrorProne) warning to use `%n` instead of `\n`. But when I use `print` method, it doesn't interpret `%n` as return code.

In addition, static analyzer recommend to use `print` method instead of `printf` method if the output string doesn't have any formatted parameter [like this](https://github.com/scalar-labs/actions/blob/add-releasenote-script/release-note-script/src/main/java/MergeReleaseNotes.java#L132).

Thus I used both `\n` and `%n`.

Reply:
Ah, I missed that `print` is used as well. Thanks!